botocore/paginate.py

```diff
@@ -64,8 +66,20 @@ def resume_token(self):
 
     @resume_token.setter
     def resume_token(self, value):
-        if isinstance(value, list):
-            self._resume_token = '___'.join([str(v) for v in value])
+        if not isinstance(value, dict):
+            raise ValueError("Bad starting token: %s" % value)
+
+        if 'truncate_amount' in value:
+            token_keys = sorted(self._input_token + ['truncate_amount'])
+        else:
+            token_keys = sorted(self._input_token)
+        dict_keys = sorted(value.keys())
+
+        if len(dict_keys) == len(token_keys) and token_keys == dict_keys:
```

Comment:
Is the length check needed? The equality check should be able to handle this already right?